CONTRIBUTING.md

```diff
@@ -67,9 +67,50 @@ If you can certify it, then just add a line to every git commit message:
 
 Use your real name (sorry, no pseudonyms or anonymous contributions).
 
-If you set your `user.name` and `user.email` git configs, you can sign your commit automatically with `git commit -s`.
-You can also use git [aliases](https://git-scm.com/book/tr/v2/Git-Basics-Git-Aliases)
-like `git config --global alias.ci 'commit -s'`. Now you can commit with `git ci` and the commit will be signed.
+### How to sign-off commits
+
+The sign-off message can either be added by hand to your commit message or automatically by git.
+
+This is accomplished by using the `-s` or `--signoff` option on your regular commit command.
+
+e.g.
+```bash
+git commit -s
+```
+or
+```bash
+git commit --signoff
+```
+
+This will use the name and email you configured in git.
+
+Find out how you can change your name and email in this guide from [GitHub](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address#setting-your-commit-email-address-in-git)
+
+#### How to amend a sign-off
+
+The last commit on any given branch can be amended to include the sign-off message like this:
+
+```bash
+git commit --amend --signoff
+```
+
+If multiple commits need a sign-off then you can use this command to sign off the entire branch:
+
+```bash
+git rebase --signoff
```

Comment:
Does this really reliably detect the right amount of commits to rebase, if the branch is already pushed?
I wonder if we should instead use `HEAD~N` to rebase the N last commits.

Reply:
The best way would be the usage of the upstream branch, preferably the remote one.

Reply:
What would you suggest adding here? A concrete commando would be useful…

Reply:
So if you use `git remote add upstream https://github.com/hedgedoc/hedgedoc.git` to add a remote upstream (as `origin` is probably the author's fork) you can do `git fetch upstream` to get the latest content from our repository and then use `git rebase --signoff upstream/master` or `git rebase --signoff upstream/develop` to rebase all commits in your current branch, that are different from upstream you try to merge with, and sign them off.